learning_resources_search/views.py

```diff
@@ -74,11 +86,97 @@ def get(self, request):
             return Response(errors, status=400)
 
 
+class UserSearchSubscriptionViewSet(viewsets.ViewSet):
+    """
+    View for listing percolate query subscriptions for a user
+    """
+
+    permission_classes = (IsAuthenticated,)
+    serializer_class = UserPercolateQueryRequestSerializer
+    filter_backends = (DjangoFilterBackend,)
+
+    @action(detail=False, methods=["POST"], name="Subscribe user to query")
+    @extend_schema(
+        summary="Subscribe user to query",
+        parameters=[LearningResourcesSearchRequestSerializer()],
+        responses=PercolateQuerySerializer(),
+    )
+    def subscribe(self, request, *args, **kwargs):  # noqa: ARG002
+        request_data = LearningResourcesSearchRequestSerializer(data=request.data)
+        if request_data.is_valid():
+            percolate_query = subscribe_user_to_search_query(
+                request.user, request_data.data | {"endpoint": LEARNING_RESOURCE}
+            )
+            return Response(PercolateQuerySerializer(percolate_query).data)
+        else:
+            errors = {}
+            for key, errors_obj in request_data.errors.items():
+                if isinstance(errors_obj, list):
+                    errors[key] = errors_obj
+                else:
+                    errors[key] = list(set(chain(*errors_obj.values())))
+            return Response(errors, status=400)
+
+    @action(
+        detail=True,
+        methods=["POST"],
+        name="Unsubscribe user from query by id",
+    )
+    @extend_schema(
+        summary="Unsubscribe user to query",
+        parameters=[LearningResourcesSearchRequestSerializer()],
+        responses=PercolateQuerySerializer(),
+    )
+    def unsubscribe(self, request, pk=None):
+        percolate_query = get_object_or_404(PercolateQuery, id=pk)
+        unsubscribe_user_from_percolate_query(request.user, percolate_query)
+        return Response(PercolateQuerySerializer(percolate_query).data)
+
+    @action(
+        detail=False,
+        methods=["POST"],
+        url_path="unsubscribe",
+        name="Unsubscribe user from query",
+    )
+    @extend_schema(
+        summary="Unsubscribe user to query",
+        parameters=[LearningResourcesSearchRequestSerializer()],
+        responses=PercolateQuerySerializer(),
+    )
+    def unsubscribe_user(self, request):
```

Comment:
I think just one of the two methods (`unsubscribe_user`, sending json in body OR `unsubscribe`, sending query id in url) is probably sufficient.  I'd defer to @ChristopherChudzicki on which one is most practical to call from the front end.

Reply:
If it's just a question of `POST /../{id}/unsubscribe` vs `POST /../unsubscribe/` + body `{ id }`, it really doesn't matter—the generated API client will treat them both as `someFunction({ id })`. 

But I agree with Matt: let's just have one.

**Also:** Can you double check the OpenAPI spec? I think some request bodies might be missing.

Reply:
just pushed.a fix for the schemas and request bodies and removed the /unsubscribe/:id route

Reply:
Just talked to @shanbady. I misunderstood the alternatives (partly due to missing openapi request bodies!).

> choice was just between:
> 1. POST /unsubscribe/{id}/ (no body)
> 2. POST /unsubscribe/ with original query in the body

IMO, (1) is preferable.

Reply:
Leaving a note here on the concensus we came to after discussing. 

- We decided to go with the /{id}/unsubscribe/ route vs POST /unsubscribe/ 
- After discussing discussing with @ChristopherChudzicki and @mbertrand I leaned towards using a DELETE instead of POST for /{id}/unsubscribe/ since from an api user's perspective we are 'deleting a subscription' along with the fact that there are no post parameters for that route , it seemed more natural to use delete